ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/ErrorsViewService.kt

```diff
@@ -50,9 +51,39 @@ class ErrorsViewService(project: Project) : AbstractViewService(project) {
         return "Errors" + if (model.errorsCount > 0) " (${model.count()})" else ""
     }
 
+
+
+    //todo: quick and dirty prevent race condition with refresh task ultil we have time to re-write it
+    fun updateErrorsModelFromRefresh(codeLessSpan: CodeLessSpan) {
+        lock.lock()
+        try {
+            //don't let the refresh task update if it's not the same CodeLessSpan
+            if ( model.scope is CodeLessSpanScope && codeLessSpan == (model.scope as CodeLessSpanScope).getSpan()) {
+                updateErrorsModelImpl(codeLessSpan)
+            }
+        }finally {
+            if (lock.isHeldByCurrentThread) {
+                lock.unlock()
+            }
+        }
+    }
+
+    //todo: quick and dirty prevent race condition with refresh task ultil we have time to re-write it
```

Comment:
what is dirty in those blocks?
can you suggest what is the proposed re-write, so we won't forget.

Reply:
its dirty because i did it in 5 minutes, i think we can achieve cleaner code with kotlin coroutines. 
I have to think about it , but in general it can be local refresh tasks that can be canceled when the context changes to a new method info.